### PR TITLE
Fixes #31 Added event and Item as param in a callback

### LINKED ISSUE #31: Can not access the same item in it's callback
I wanted to access some properties of my item in its callback for further use but it's not available as it's not been passed
Even the event object is not passed which we get with the `onClick` attribute

I solved this issue in PR #30 by passing the event and the item to the callback in that order i.e. `(event, item)`
I hope it's helpful

### PR DESCRIPTION
**Pull request by YASHBRO**
Fixes #31 
Added: Event and Item as param in a callback
Updated: type of callback in their respective interface

### CODE REVIEW

**Comment by steviebaa** [comment]
Thanks for the PR @YASHBRO.

It looks like your formatter changed quite a bit - my fault for not including a `.prettierrc`.

I've added a prettier config file to the project, formatted everything and pushed it to main. Can you rebase your branch with main and just check there isn't unnecessary formatting changes in the PR.

Thanks

**Comment by YASHBRO** [comment]
Hey @steviebaa 

I have rebased my branch with the main branch
Now in the _files changed_ tab, there are only necessary changes showing up

Let me know if there's any other change you want me to implement

File: apps/docs/src/pages/NestedDropdownPage.tsx [hunk 1]
```diff
@@ -32,12 +38,18 @@ const menuItemsData: MenuItemData = {
 				{
 					label: 'Option 1',
 					rightIcon: <SaveAsIcon />,
-					callback: () => console.log('Save As > Option 1 clicked'),
+					callback: (
+						event: React.MouseEvent<HTMLElement>,
+						item: MenuItemData
+					) => console.log('Save As > Option 1 clicked'),
```
Comment (steviebaa) [inline apps/docs/src/pages/NestedDropdownPage.tsx:44]: add `event, item` to the log for consistency

File: apps/docs/src/pages/NestedDropdownPage.tsx [hunk 2]
```diff
@@ -32,12 +38,18 @@ const menuItemsData: MenuItemData = {
...
 				},
 				{
 					label: 'Option 2',
 					leftIcon: <SaveAsIcon />,
-					callback: () => console.log('Save As > Option 2 clicked'),
+					callback: (
+						event: React.MouseEvent<HTMLElement>,
+						item: MenuItemData
+					) => console.log('Save As > Option 2 clicked'),
```
Comment (steviebaa) [inline apps/docs/src/pages/NestedDropdownPage.tsx:52]: add `event, item` to the log for consistency

File: apps/docs/src/pages/NestedDropdownPage.tsx [hunk 3]
```diff
@@ -53,21 +65,28 @@ const menuItemsData: MenuItemData = {
 						{
 							label: 'Option 1',
 							rightIcon: <SaveAsIcon />,
-							callback: () =>
-								console.log('Export > FT1 > O1 clicked'),
+							callback: (
+								event: React.MouseEvent<HTMLElement>,
+								item: MenuItemData
+							) => console.log('Export > FT1 > O1 clicked'),
```
Comment (steviebaa) [inline apps/docs/src/pages/NestedDropdownPage.tsx:71]: add `event, item` to the log for consistency

File: apps/docs/src/pages/NestedDropdownPage.tsx [hunk 4]
```diff
@@ -53,21 +65,28 @@ const menuItemsData: MenuItemData = {
...
 						},
 						{
 							label: 'Option 2',
 							leftIcon: <SaveAsIcon />,
-							callback: () =>
-								console.log('Export > FT1 > O2 clicked'),
+							callback: (
+								event: React.MouseEvent<HTMLElement>,
+								item: MenuItemData
+							) => console.log('Export > FT1 > O2 clicked'),
```
Comment (steviebaa) [inline apps/docs/src/pages/NestedDropdownPage.tsx:80]: add `event, item` to the log for consistency

File: apps/docs/src/pages/NestedDropdownPage.tsx [hunk 5]
```diff
@@ -53,21 +65,28 @@ const menuItemsData: MenuItemData = {
...
 						},
 					],
 				},
 				{
 					label: 'File Type 2',
-					callback: () => console.log('Export > FT2 clicked'),
+					callback: (
+						event: React.MouseEvent<HTMLElement>,
+						item: MenuItemData
+					) => console.log('Export > FT2 clicked'),
```
Comment (steviebaa) [inline apps/docs/src/pages/NestedDropdownPage.tsx:89]: add `event, item` to the log for consistency

File: packages/mui-nested-menu/src/definitions.ts [hunk 6]
```diff
@@ -1,12 +1,12 @@
-import { SxProps } from '@mui/system/styleFunctionSx/styleFunctionSx';
+import { SxProps } from "@mui/system/styleFunctionSx/styleFunctionSx";
```
Comment (steviebaa) [inline packages/mui-nested-menu/src/definitions.ts:1]: revert the changes to quotes.

File: packages/mui-nested-menu/src/definitions.ts [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
 export interface MenuItemData {
-	uid?: string;
-	label?: string;
-	leftIcon?: React.ReactNode;
-	rightIcon?: React.ReactNode;
-	callback?: () => void;
-	items?: MenuItemData[];
-	disabled?: boolean;
-	sx?: SxProps;
+    uid?: string;
```
Comment (steviebaa) [inline packages/mui-nested-menu/src/definitions.ts:4]: revert the changes to indentation.

**Review by steviebaa — CHANGES_REQUESTED** [review]
Just a few minor changes. 

Another note, when clicking the New item in the first dropdown on the [preview page](https://mui-nested-menu-6zkzw2lko-steviebaa.vercel.app/nested-dropdown), I am seeing undefined. Did you test that the event and items were coming through?

Screenshot 2023-01-14 at 08 35 42

**Comment by YASHBRO** [comment]
Sorry, @steviebaa for the delay.
I am occupied with my exams, I'll resolve these as soon as I'm done with them, that'll take 2-3 days.

**Comment by steviebaa** [comment]
Hey @YASHBRO are you able to wrap this one up?

I will try to do a sweep on the codebase sometime soon to clean it up and it will likely create a few conflicts.

If you don't have time then I can finish it up but I'm not sure you'll get the credit as I'll need to make a new branch.

**Comment by YASHBRO** [comment]
Hello @steviebaa 
Sorry, it slipped out of my mind.
I'll wrap it up by tonight

**Comment by YASHBRO** [comment]
> Just a few minor changes @YASHBRO
> 
> Another note, when clicking the New item in the first dropdown on the [preview page](https://mui-nested-menu-6zkzw2lko-steviebaa.vercel.app/nested-dropdown), I am seeing undefined. Did you test that the event and items were coming through?
> 
> Screenshot 2023-01-14 at 08 35 42

I am not able to replicate this error in my local machine, it works fine in my local machine but gives `undefined` in this preview link

**Comment by steviebaa** [comment]
Thanks for getting this done Yash.

**Pull request merged**